### PR TITLE
new option: "Silent error on save"

### PR DESCRIPTION
**Pull request by oriSomething**
Continuation of #45 

Please review the code. I'm sure you might want to solve this problem differently

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -58,47 +58,54 @@
       "default": false,
       "order": 1
     },
+    "formatOnSaveSilentError": {
+      "title": "On format on Save, Don't show error message",
+      "description": "Don't show error if accour on format on save",
```
Comment (robwise) [inline package.json]: Typo? `accour`

File: package.json [hunk 2]
```diff
@@ -58,47 +58,54 @@
       "default": false,
       "order": 1
     },
+    "formatOnSaveSilentError": {
+      "title": "On format on Save, Don't show error message",
```
Comment (robwise) [inline package.json]: There's some inconsistent capitalization going on here

File: lib/helpers.js [hunk 3]
```diff
@@ -94,7 +93,12 @@ const formatOnSaveIfEnabled = () => {
     return;
   }
 
-  format(null, { ignoreSelection: true });
+  const formatOnSaveSilentError = getConfigOption('formatOnSaveSilentError');
+
+  format(null, {
+    ignoreSelection: true,
+    ignoreErrors: !formatOnSaveSilentError,
```
Comment (robwise) [inline lib/helpers.js]: I would just inline the call to `getConfigOption`, or at least name the variablje `ignoreErrors` so you can just use property shorthand

File: lib/helpers.js [hunk 4]
```diff
@@ -62,22 +61,22 @@ const format = (event, options = { ignoreSelection: false }) => {
...
     } else {
-      transformRange(editor, bufferRange);
+      transformRange(editor, bufferRange, { ignoreErrors: options.ignoreErrors });
     }
   } else {
     // transform selected ranges
-    editor.getSelectedBufferRanges().forEach(range => {
-      transformRange(editor, range);
+    editor.getSelectedBufferRanges().forEach((range) => {
+      transformRange(editor, range, { ignoreErrors: options.ignoreErrors });
```
Comment (robwise) [inline lib/helpers.js]: we should just pass `options` instead of redefining a new object every time like this

File: lib/helpers.js [hunk 5]
```diff
@@ -62,22 +61,22 @@ const format = (event, options = { ignoreSelection: false }) => {
...
         const { start, end } = iter.range;
-        const startModified = [start.row + 1, start.column];
+        const startModified = [ start.row + 1, start.column ];
         const range = new iter.range.constructor(startModified, end);
-        transformRange(editor, range);
+        transformRange(editor, range, { ignoreErrors: options.ignoreErrors });
       });
     } else {
-      transformRange(editor, bufferRange);
+      transformRange(editor, bufferRange, { ignoreErrors: options.ignoreErrors });
```
Comment (robwise) [inline lib/helpers.js]: we should just pass `options` instead of redefining a new object every time like this

File: lib/helpers.js [hunk 6]
```diff
@@ -62,22 +61,22 @@ const format = (event, options = { ignoreSelection: false }) => {
...
+      editor.backwardsScanInBufferRange(re, bufferRange, (iter) => {
         // Create new range with start row advanced by 1,
         // since we cannot use look-behind on variable-length starting
         // <script ...> tag
         const { start, end } = iter.range;
-        const startModified = [start.row + 1, start.column];
+        const startModified = [ start.row + 1, start.column ];
         const range = new iter.range.constructor(startModified, end);
-        transformRange(editor, range);
+        transformRange(editor, range, { ignoreErrors: options.ignoreErrors });
```
Comment (robwise) [inline lib/helpers.js]: we should just pass `options` instead of redefining a new object every time like this

File: lib/helpers.js [hunk 7]
```diff
@@ -24,22 +21,24 @@ const getFormatOptions = () => PRETTIER_CORE_OPTIONS.reduce(setConfigOption, {})
 
 const getCurrentScope = () => atom.workspace.getActiveTextEditor().getGrammar().scopeName;
 
-const executePrettier = (text, formatOptions) => {
+const executePrettier = (text, formatOptions, options = { ignoreErrors: false }) => {
   try {
     return prettier.format(text, formatOptions);
   } catch (e) {
     const message = `prettier-atom: ${e.toString()}`;
```
Comment (robwise) [inline lib/helpers.js]: we should use a guard clause instead that way it's clear what code goes with what:

```js
  console.log('Error executing Prettier:', e);

  if (!options.ignoreErrors) {
    const message = `prettier-atom: ${e.toString()}`;
    const detail = e.stack.toString();
    atom.notifications.addError(message, { detail, dismissable: true });
  }
  
  return false;
}
```

File: lib/helpers.js (hunk 4 above)
Comment (oriSomething) [inline lib/helpers.js]: But the options of `format` contains options which aren't related to `transformRange`. I prefer to pass to a function only properties that are relevant to that function

**Comment by oriSomething** [comment]
By the way tell when code is ready so I'll rebase history it before merge

File: lib/helpers.js (hunk 4 above)
Comment (robwise) [inline lib/helpers.js]: ok fair enough

**Comment by oriSomething** [comment]
Done

**Pull request merged**